src/api/SFA.DAS.Provider.Events.Application.UnitTests/Submissions/GetLatestLearnerEventByStandardQuery/GetLatestLearnerEventByStandardQueryHandler/WhenHandling.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Moq;
+using NUnit.Framework;
+using SFA.DAS.Provider.Events.Application.Data.Entities;
+using SFA.DAS.Provider.Events.Application.Repositories;
+using SFA.DAS.Provider.Events.Application.Submissions.GetLatestLearnerEventByStandardQuery;
+using SFA.DAS.Provider.Events.Application.Validation;
+
+namespace SFA.DAS.Provider.Events.Application.UnitTests.Submissions.GetLatestLearnerEventByStandardQuery.GetLatestLearnerEventByStandardQueryHandler
+{
+    public class WhenHandling
+    {
+        private Mock<IValidator<GetLatestLearnerEventByStandardQueryRequest>> _validator;
+        private Mock<ISubmissionEventsRepository> _submissionEventsRepository;
+        private Application.Submissions.GetLatestLearnerEventByStandardQuery.GetLatestLearnerEventByStandardQueryHandler _handler;
+        private GetLatestLearnerEventByStandardQueryRequest _request;
+
+        [SetUp]
+        public void Arrange()
+        {
+            _validator = new Mock<IValidator<GetLatestLearnerEventByStandardQueryRequest>>();
+            _validator.Setup(v => v.Validate(It.IsAny<GetLatestLearnerEventByStandardQueryRequest>()))
+                .ReturnsAsync(new ValidationResult());
+
+            _submissionEventsRepository = new Mock<ISubmissionEventsRepository>();
+
+            _handler = new Application.Submissions.GetLatestLearnerEventByStandardQuery.GetLatestLearnerEventByStandardQueryHandler(
+                _validator.Object, _submissionEventsRepository.Object);
+
+            _request = new GetLatestLearnerEventByStandardQueryRequest();
+        }
+
+        [Test]
+        public void ThenItShouldReturnInvalidResponseWhenValidatorFails()
+        {
+            _validator.Setup(v => v.Validate(It.IsAny<GetLatestLearnerEventByStandardQueryRequest>()))
+                .ReturnsAsync(new ValidationResult
+                {
+                    ValidationMessages = new[] { "Invalid" }
+                });
+
+            // Act
+            var actual = _handler.Handle(_request).Result;
+
+            // Assert
+            Assert.IsNotNull(actual);
+            Assert.IsFalse(actual.IsValid);
+            Assert.IsInstanceOf<ValidationException>(actual.Exception);
+        }
+
+        [Test]
+        public void ThenItShouldReturnInvalidResponseWhenAnExceptionIsThrown()
+        {
+            _validator.Setup(v => v.Validate(It.IsAny<GetLatestLearnerEventByStandardQueryRequest>()))
+                .Throws<Exception>();
+
+            // Act
+            var actual = _handler.Handle(_request).Result;
+
+            // Assert
+            Assert.IsNotNull(actual);
+            Assert.IsFalse(actual.IsValid);
+            Assert.IsInstanceOf<Exception>(actual.Exception);
+        }
+
+        [Test]
+        public void ThenItShouldReturnValidResults()
+        {
+            _submissionEventsRepository.Setup(r => r.GetLatestLearnerEventByStandard(It.IsAny<long>(), It.IsAny<long>()))
```

Comment:
Can It.IsAny be changed to the actual expected parameters?  It'll make the test a lot tighter.